lib/lte_link_control/lte_lc.c

```diff
@@ -163,6 +177,22 @@ void at_handler(void *context, const char *response)
 
 static int w_lte_lc_init(void)
 {
+#if defined(CONFIG_LWM2M_CARRIER)
+#if !defined(CONFIG_GPS_USE_SIM) && \
+	(defined(CONFIG_BOARD_NRF9160_PCA20035NS) || \
+	 defined(CONFIG_BOARD_NRF9160_PCA10090NS))
+	/* Configuring MAGPIO/COEX, so that the correct antenna
+	 * matching network is used for each LTE band and GPS.
+	 */
+	for (size_t i = 0; i < ARRAY_SIZE(lwm2m_ant_cfg); i++) {
+		if (at_cmd_write(lwm2m_ant_cfg[i], NULL, 0, NULL) != 0) {
+			return -EIO;
+		}
+	}
+#endif
+	return 0;
```

Comment:
Do we want to return here?

Reply:
yes i believe so, previously if LWM2M was enable, `lte_lc_init_and_connect` didn't even get called.

Reply:
Oh now I see. That's because the carrier lib calls these functions, which is probably not documented. So I think they will need to continue execution here.
@stig-bjorlykke please correct me if I'm wrong here

Reply:
ah, if that is the case i will update main.c to not call `lte_lc_init`.

Reply:
confirmed via log message that something is calling `lte_lc_init`.
updated.